### ISSUE TITLE
Inconsistency between `mix` and `average`

### ISSUE THREAD

**Issue by jiho** [gka/chroma.js#176]
I understood averaging as a more general version of mixing, for more than two colors, but:

```javascript
clrs=[chroma.random(),chroma.random()]

// in those two modes, mix and average give the same result
chroma.mix(clrs[0], clrs[1], 0.5, 'rgb').hex()
chroma.average(clrs, 'rgb').hex()

chroma.mix(clrs[0], clrs[1], 0.5, 'lab').hex()
chroma.average(clrs, 'lab').hex()

// in the following they don't
chroma.mix(clrs[0], clrs[1], 0.5, 'hcl').hex()
chroma.average(clrs, 'hcl').hex()

chroma.mix(clrs[0], clrs[1], 0.5, 'hsv').hex()
chroma.average(clrs, 'hsv').hex()

chroma.mix(clrs[0], clrs[1], 0.5, 'hsl').hex()
chroma.average(clrs, 'hsl').hex()

chroma.mix(clrs[0], clrs[1], 0.5, 'hsi').hex()
chroma.average(clrs, 'hsi').hex()

chroma.mix(clrs[0], clrs[1], 0.5, 'lrgb').hex()
chroma.average(clrs, 'lrgb').hex()
```

**Comment by danburzo** [comment]
The bulk of the 'inconsistency' is that mix() and average() treat the Hue component differently. mix() interpolates between the hues by taking the _shortest_ distance between them, while average returns their average. (e.g. 350 and 10 are only 20 degrees apart when interpolating them, and their average is 180). As for LRGB, interpolating and averaging are not equivalent.

**Comment by jiho** [comment]
Thanks for the quick response (again!). So it seems there are several issues mixed together:

- first, and easy to fix, [in the docs](https://gka.github.io/chroma.js/#chroma-average), `chroma.average` is described as "Similar to chroma.mix, but accepts more than two colors." From that sentence, I would have expected them to be the same when used with just two colors. So it may be a matter of just fixing the docs.

- second, the docs again state "Simple averaging of R,G,B components and the alpha channel", which makes sense for RGB but does not for anything with a hue. I would argue that taking the arithmetic average of angles is just wrong. The average between `hcl(10, 30, 20)` and `hcl(350, 40, 10)` should be `hcl(0, 35, 15)` *not* `hcl(180, 35, 15)`. Rosetta code has a [Javascript implementation of the mean of angles](https://rosettacode.org/wiki/Averages/Mean_angle#JavaScript).

- third, regarding LRGB... I have no clue ;-)

**Comment by danburzo** [comment]
(Sorry, I have deleted my inaccurate answer about hue averaging in the meantime, I had misread the code) Seems the hue average was incorrectly averaged up until two hours ago :) https://github.com/gka/chroma.js/pull/164 As for the LRGB, I'll be curious to see what's going on, because on a  closer read they should match!

**Comment by gka** [comment]
thanks, I will take a look

**Comment by gka** [comment]
'green' is exactly `#008800` so this must be a bug in chroma.average then.

**Comment by gka** [comment]
this bug is fixed in chroma 2.0.0
